### PR TITLE
Add publish families to cache creators

### PR DESCRIPTION
**Pull request by MustafaJafar**
## Changelog Description
This PR 
- adds  publish families to cache creators.
- removes unnecessary collector.
- implements callbacks to cachesize collector to make its attribute visible if `farm`  is enabled

> [!Note]
> This PR is mainly made for sharing code and show a use case  for https://github.com/ynput/ayon-core/pull/994
> It should be used along https://github.com/ynput/ayon-deadline/pull/49 and https://github.com/ynput/ayon-deadline/pull/54
> Here's what to expect when using the three PRs togther
> Animation_17

## Testing notes:
1. Launch Houdini via AYON
2. Create a cache instance (vdb, model, pointcache, ass, rsproxy)
3. Switch farm toggle in the publisher UI, click save,  click refresh and watch the attributes appear and disappear.

### CODE REVIEW

File: client/ayon_houdini/plugins/publish/collect_chunk_size.py
Comment (BigRoy) [inline client/ayon_houdini/plugins/publish/collect_chunk_size.py:1]: Honestly this plug-in should never have been in ayon-houdini to begin with. It's related to farm setting and this should really be in `ayon-deadline` as a setting most likely for e.g. a `farm` family.

File: client/ayon_houdini/plugins/publish/collect_instances_type.py
Comment (BigRoy) [inline client/ayon_houdini/plugins/publish/collect_instances_type.py:1]: You got to love some good cleanup. ❤️ 🥳

File: client/ayon_houdini/plugins/publish/collect_chunk_size.py
Comment (MustafaJafar) [reply client/ayon_houdini/plugins/publish/collect_chunk_size.py:1]: yeah, totally agreed.

Comment (MustafaJafar) [reply client/ayon_houdini/plugins/publish/collect_chunk_size.py:1]: I'm going to mention it in the houdini deadline refactoring issue https://github.com/ynput/ayon-houdini/issues/33

**Review by moonyuet — COMMENTED** [review]
I did simple test on this, it is submitted to the farm successfully
image
Are we gonna wait for the merge of this PR https://github.com/ynput/ayon-houdini/pull/171 ? Or there should be something needed to be working on as I see this PR is still a draft?

**Comment by MustafaJafar** [comment]
> Are we gonna wait for the merge of this PR #171 ?

Both are separate PRs. they don't depend on each other.

**Comment by MustafaJafar** [comment]
I can still submit to deadline when I'm on this PR but having `publish.hou` family in `get_publish_families` has a downside, that deadline plugins will still be triggered even if I don't have farm enabled. 

This plugin shouldn't be triggered when disabling submit to farm. 
image

**Comment by MustafaJafar** [comment]
This PR shouldn't add the intermediate publish family `publish.hou` used for submitting to deadline.
as it will always assume they are farm instances.

**Comment by BigRoy** [comment]
Actually @MustafaJafar - I think it is up to the creators to do so. This PR wasn't too far off, it should just be more dynamic.

As soon as you toggle (on the creator) to submit it to the farm or not - then the relevant families should be added if set to farm.
Then on the deadline plug-in it should respond to changes on `instance["families"]` so that if the `farm.publish` (or "publish.hou" if you prefer that naming) job is not set that it's not active.

That way the attributes are still completely aligned with the instance's family.

**Comment by MustafaJafar** [comment]
Is that achieved by implementing a callback that listens to the value change and assigns the family ?

**Comment by BigRoy** [comment]
> Is that achieved by implementing a callback that listens to the value change and assigns the family ?

Correct - value change callback on the creator which adds/removes the farm related family based on the enum's state.

One could approach it the opposite so that all this logic is in the Deadline plug-in to watch for creator attribute changes, however that means it's up to that plug-in to detect whether "farm is enabled" on the attribute which makes it harder to maintain over time if we want to expose the toggle in the creator a different way. Or like add a different additional toggle or whatever.

**Pull request closed without merging**